backend/src/handlers/facilities_handler.go

```diff
@@ -16,14 +16,28 @@ func (srv *Server) registerFacilitiesRoutes() {
 	srv.Mux.Handle("PUT /api/admin/facility-context/{id}", srv.applyAdminMiddleware(srv.handleChangeAdminFacility))
 }
 
+/**
+* GET: /api/facility/{id}
+**/
 func (srv *Server) handleIndexFacilities(w http.ResponseWriter, r *http.Request, log sLog) error {
-	facilities, err := srv.Db.GetAllFacilities()
+	page, perPage := srv.getPaginationInfo(r)
+	total, facilities, err := srv.Db.GetAllFacilities(page, perPage)
 	if err != nil {
 		return newDatabaseServiceError(err)
 	}
-	return writeJsonResponse(w, http.StatusOK, facilities)
+	last := srv.calculateLast(total, perPage)
+	paginationData := models.PaginationMeta{
+		PerPage:     perPage,
+		LastPage:    int(last),
+		CurrentPage: page,
+		Total:       total,
+	}
```

Comment:
Just noticed this, but you can actually use this to do this directly:
`paginationData := models.NewPaginationInfo(page, perPage, total)`

Reply:
so we can get rid of the variable used for LastPage altogether, meaning remove line 28 as well?

Reply:
Yes, I believe so